server/main.lua

```diff
@@ -0,0 +1,66 @@
+---@class InsertVehicleEntityQuery
+---@field license string
+---@field citizenId string
+---@field model string
+---@field plate string
+---@field mods string
+
+--- Creates a Vehicle DB Entity
+---@param query InsertVehicleEntityQuery Insert Vehicle Entity Request
+local function createEntity(query)
+    MySQL.insert('INSERT INTO player_vehicles (license, citizenid, vehicle, hash, mods, plate, state) VALUES (?,?,?,?,?,?,?)', {
+        query.license,
+        query.citizenId,
+        query.model,
+        joaat(query.model),
+        query.mods or '{}',
+        query.plate,
+        0
+    })
+end
+
+exports('CreateEntity', createEntity)
+
+---@class FetchVehicleEntityQuery
+---@field valueType 'citizenid'|'license'|'plate'
+---@field value string
+
+---@alias vehicleEntity table
+
+--- Fetches DB Vehicle Entity
+---@param query FetchVehicleEntityQuery
+---@return vehicleEntity[]
+local function fetchEntity(query)
+    if query.valueType ~= 'citizenid' and query.valueType ~= 'license' and query.valueType ~= 'plate' then return end
+    return MySQL.await('SELECT * FROM player_vehicles WHERE ? = ?', {
+        query.valueType,
+        query.value
+    })
+end
+
+exports('FetchEntity', fetchEntity)
+
+---@class UpdateEntityVehicleQuery
+---@field valueType string
+---@field value string
+
+--- Updates a DB Vehicle Entity
+---@param query UpdateEntityVehicleQuery
+---@param vehiclePlate string
+local function updateEntity(query, vehiclePlate)
+    MySQL.update('UPDATE player_vehicles SET ? = ?, WHERE plate = ?', {
+        query.valueType,
+        query.value,
+        vehiclePlate
+    })
+end
```

Comment:
This exposes updating any value. However, looking at the vehicle values,  do we want model and hash changed? I would think about the use cases and derive API from there. For example, if a player transfers a vehicle, what DB changes need to be made. If a player puts a vehicle in impound/garage/out what DB changes? Sometimes this requires updating one field, other use cases may require updating multiple. I don't think you can have a one size fits all function here. You may need a couple of targeted queries to hit all the use cases.

Reply:
I won't touch garages, as the issue suggests that qbx_garages should handle that. But I've added a new function to change the owner, as that requires both the citizenId and the license to change. Through what I see, that seems to be the only thing that requires an explicit function.

I think swapping out the updateEntityFunction with a string builder might be a way to go there.